librz/il/definitions/bitvector.c

```diff
@@ -1111,7 +1125,13 @@ RZ_API bool rz_il_bv_set_from_st32(RZ_NONNULL RzILBitVector *bv, st32 value) {
  * \param value st64, the value to convert
  */
 RZ_API bool rz_il_bv_set_from_st64(RZ_NONNULL RzILBitVector *bv, st64 value) {
-	rz_return_val_if_fail(bv && bv->bits, false);
+	rz_return_val_if_fail(bv, false);
+	if (bv->len <= 64) {
+		bv->bits.small_s = value;
+		bv->bits.small_u &= (UT64_MAX >> (64 - bv->len));
```

Comment:
@thestr4ng3r it does work in the way it should. check this.
i also have added tests on purpose for this.
at C level negative numbers does indeed have 63rd upper bit there for signess, but i use this only for conversion to unsigned and then end.
when your number is >= `0x8000 0000 0000 0000` is considered as negative, but when you give as input a number (let's say `-100`) which only needs 5 bits, all the upper bits are 1 therefore you can do this magic by masking out the bits and still get the right upper bits as 1 (i clear the upper bits on purpose)